src/app/pages/back-to-office/__tests__/back-to-office.page.spec.ts

```diff
@@ -1,31 +1,170 @@
-import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
-import { IonicModule, NavController } from '@ionic/angular';
-import { NavMock } from '@mocks/angular-mocks/nav-mock';
+import { ComponentFixture, waitForAsync, TestBed } from '@angular/core/testing';
+import {
+  IonicModule, NavParams, Config, Platform,
+} from '@ionic/angular';
+import {
+  NavParamsMock, ConfigMock, PlatformMock,
+} from 'ionic-mocks';
+
+import { AppModule } from 'src/app/app.module';
+import { AuthenticationProvider } from '@providers/authentication/authentication';
+import { AuthenticationProviderMock } from '@providers/authentication/__mocks__/authentication.mock';
+import { DateTimeProvider } from '@providers/date-time/date-time';
+import { DateTimeProviderMock } from '@providers/date-time/__mocks__/date-time.mock';
+import { StoreModule, Store } from '@ngrx/store';
+import { StoreModel } from '@shared/models/store.model';
+import { ScreenOrientation } from '@ionic-native/screen-orientation/ngx';
+import { Insomnia } from '@ionic-native/insomnia/ngx';
+import { DeviceProvider } from '@providers/device/device';
+import { DeviceProviderMock } from '@providers/device/__mocks__/device.mock';
+import { InsomniaMock } from '@shared/mocks/insomnia.mock';
+import { ScreenOrientationMock } from '@shared/mocks/screen-orientation.mock';
+import { MockComponent } from 'ng-mocks';
+import { PracticeModeBanner } from '@components/common/practice-mode-banner/practice-mode-banner';
+import { By } from '@angular/platform-browser';
+import { of } from 'rxjs';
+import { configureTestSuite } from 'ng-bullet';
+import { Router } from '@angular/router';
 import { RouteByCategoryProvider } from '@providers/route-by-category/route-by-category';
 import { RouteByCategoryProviderMock } from '@providers/route-by-category/__mocks__/route-by-category.mock';
-
 import { BackToOfficePage } from '../back-to-office.page';
 
 describe('BackToOfficePage', () => {
-  let component: BackToOfficePage;
   let fixture: ComponentFixture<BackToOfficePage>;
+  let component: BackToOfficePage;
+  let store$: Store<StoreModel>;
+  let screenOrientation: ScreenOrientation;
+  let insomnia: Insomnia;
+  let deviceProvider: DeviceProvider;
+  let router: Router;
 
-  beforeEach(waitForAsync(() => {
+  configureTestSuite(() => {
     TestBed.configureTestingModule({
-      declarations: [BackToOfficePage],
-      imports: [IonicModule.forRoot()],
+      declarations: [
+        BackToOfficePage,
+        MockComponent(PracticeModeBanner),
+      ],
+      imports: [
+        IonicModule,
+        AppModule,
+        StoreModule.forRoot({
+          tests: () => ({
```

Comment:
Possibly not something for this ticket, but the mock here of the store data from memory appears in multiple places. Could we possibly add a TODO to look at centralising this mock object?

Reply:
have removed the object as it's actually not even used here